### DISCUSSION TITLE
Get coords of a block pointed by the player

### DISCUSSION THREAD

**Discussion by UranusDarkness** [FabricMC/fabric-api#3165] [Mod Dev Support]
### Description

Hello! Is it possible to get the coords of a Block pointed by the Player? 
For example: the player is pointing at a chest, and I want to get the chest's coords 
I am creating a server-sided mod, so it must work on server-side as well

**Comment by haykam821** [comment]
In Minecraft 1.20.1, you can use a raycast to get the currently targeted block position:

```java
public static BlockPos getTargetedBlockPos(ServerPlayerEntity player) {
	double maxDistance = player.isCreative() ? 5 : 4.5;
	HitResult hit = player.raycast(maxDistance, 0, false);

	if (hit instanceof BlockHitResult blockHit) {
		return blockHit.getBlockPos();
	}
}

**Reply by lafkpages** [reply]
Do you know what the second argument to `raycast()` does?
